packages/gitmoji-changelog-core/src/index.js

```diff
@@ -45,43 +50,91 @@ function makeGroups(commits) {
     .filter(group => group.commits.length)
 }
 
-async function generateChangelog() {
-  const packageInfo = await getPackageInfo()
-  const repository = await getRepoInfo(packageInfo)
+function sanitizeVersion(version) {
+  return semver.valid(version, {
+    loose: false,
+    includePrerelease: true,
+  })
+}
 
-  const meta = {
-    package: packageInfo,
-    repository,
+async function generateChanges({ from, to, version }) {
+  const commits = await getCommits(from, to)
+  const lastCommitDate = getLastCommitDate(commits)
+
+  return {
+    version,
+    date: version !== 'next' ? lastCommitDate : undefined,
+    groups: makeGroups(commits),
   }
+}
 
+async function generateTagsChanges(tags) {
   let previousTag = ''
+
+  return Promise.all(tags.map(async tag => {
+    const changes = await generateChanges({
+      from: previousTag,
+      to: tag,
+      version: sanitizeVersion(tag),
+    })
+    previousTag = tag
+    return changes
+  })).then((changes) => {
+    return changes.sort((c1, c2) => semverCompare(c1.version, c2.version)).reverse()
```

Comment:
You don't have to reverse if you reverse changes in compare fonction, non ?

Reply:
Yes but it seems less clear, no ?

```js
changes.sort((c1, c2) => -semverCompare(c1.version, c2.version))
```

Reply:
As you prefer :wink: